storage/range.go

```diff
@@ -625,11 +638,27 @@ func (r *Range) addReadOnlyCmd(ctx context.Context, args proto.Request, reply pr
 // write is removed from the read queue and the reply is added to the
 // response cache.  If wait is true, will block until the command is
 // complete.
-func (r *Range) addWriteCmd(ctx context.Context, args proto.Request, reply proto.Response, wait bool) error {
-	// Check the response cache in case this is a replay. This call
-	// may block if the same command is already underway.
+func (r *Range) addWriteCmd(ctx context.Context, args proto.Request, reply proto.Response, wg *sync.WaitGroup) error {
+	var once sync.Once
+	signal := func() {
+		once.Do(func() {
```

Comment:
`once.Do` is a little heavyweight for this since it has thread-safety we don't need (and additionally optimizes the "already done" case at the expense of the first-time case). I think we can just do this:

```
signal := func() {
    if wg != nil {
        wg.Done()
        wg = nil
    }
}
```